discord/http.py

```diff
@@ -260,123 +238,135 @@ async def request(
         if self.proxy_auth is not None:
             kwargs["proxy_auth"] = self.proxy_auth
 
-        if not self._global_over.is_set():
-            # wait until the global lock is complete
-            await self._global_over.wait()
+        if self.global_dynamo and self.global_dynamo.rate_limited:
+            await self.global_dynamo.wait()
+
+        tbucket = await self._rate_limit.temp_bucket(bucket_id)
 
-        response: aiohttp.ClientResponse | None = None
-        data: dict[str, Any] | str | None = None
-        await lock.acquire()
-        with MaybeUnlock(lock) as maybe_lock:
+        if tbucket:
+            await tbucket.wait()
+
+        async with self._rate_limit.global_concurrency:
```

Comment:
Shouldn't the global concurrency be waited for each request try made? Currently it only checks once, when a single HTTP client request call can make up to 5 requests before failing.